health/checker.go

```diff
@@ -59,12 +59,11 @@ func (c *checker) Check(ctx context.Context) (*Health, error) {
 	}
 	healthy := true
 	for _, dep := range c.deps {
-		err := dep.Ping(ctx)
 		res.Status[dep.Name()] = "OK"
-		if err != nil {
+		if err := dep.Ping(ctx); err != nil {
 			res.Status[dep.Name()] = "NOT OK"
 			healthy = false
-			log.Error(ctx, err, log.KV{"msg", "ping failed"}, log.KV{"target", dep.Name()})
+			log.Error(ctx, err, log.KV{K: "msg", V: "ping failed"}, log.KV{K: "target", V: dep.Name()})
 		}
 	}
 	var err error
```

Comment:
Wouldn't the signature of `Check()` be better in the following form?

```go
Check(context.Context) (*Health), bool)
```

Reply:
Is there any possibilities that `Check()` will return an error other than `ErrUnhealthy` in the future?
I think it's enough to return healthy or unhealthy as bool.

Reply:
Oh sorry my mistake, was thinking of the `Pinger` interface - not  `Check`. Yes you are right that this could be a boolean, want to make the change?

Reply:
Yes, I'll create another pull request. I would be happy to merge this one.